[docs]Describe file upload procedure (#5525)
DevExpress/testcafe: docs/articles/documentation/reference/test-api/testcontroller/setfilestoupload.md

---
# t.setFilesToUpload Method

> Populates the specified file upload element with file paths. The element must be an `<input>` with the `type="file"` attribute.

Reviewer: > Populates the specified file upload element with file paths.

Emulate the choosing specified files to upload with browser's 'Choose File' dialog.

Author: > Selects files to upload. Emulates a user's selection in the browser's 'Choose File' dialog.

Parameter  | Type                                              | Description
---------- | ------------------------------------------------- | -------------------------------------------------------------------------------------------------------------------------------------------
`selector` | Function &#124; String &#124; Selector &#124; Snapshot &#124; Promise | Identifies the input field to which file paths are written. See [Select Target Elements](#select-target-elements).

> `filePath` | String &#124; Array                                            | The path to the uploaded file, or multiple such paths. Relative paths are resolved against the folder with the test file.

Reviewer 2: ```suggestion
`filePath` | String &#124; Array                                            | The path(s) to the uploaded file. Relative paths are resolved against the folder with the test file.
```
Relative paths - what does relative mean here?

The `t.setFilesToUpload` action removes all file paths from the input before populating it with new ones.

> > If an error occurs while uploading files, the test fails.

Reviewer 2: ```suggestion
> If an error occurs while files are being uploaded, the test fails.
```

> > If an error occurs while files are being uploaded, the test fails.

Reviewer 2: ```suggestion
> If an error occurs while files are being uploaded, the test will fail.
```
